### PR TITLE
Switch port

### PR DESCRIPTION
**Pull request by paulsapps**

### CODE REVIEW

File: Source/AliveLib/MainMenu.cpp [hunk 1]
```diff
@@ -2062,8 +2064,15 @@ signed int MainMenuController::LoadDemo_Update_4D1040(DWORD)
         strcpy(&lvFilename[7], sPathData_559660.paths[levelId].field_22_lvl_name_cd);
         auto lvFilenameNoPrefix = &lvFilename[7];
 
-        while (!MainMenuController::checkIfDemoFileExists_4D1430(lvFilenameNoPrefix) && !MainMenuController::checkIfDemoFileExists_4D1430(lvFilename))
+#if BEHAVIOUR_CHANGE_SUB_DATA_FOLDERS
+		std::string subFolderPath(&lvFilenameNoPrefix[1]);
+		subFolderPath = "levels\\" + subFolderPath;
+        while (!MainMenuController::checkIfDemoFileExists_4D1430((char*)subFolderPath.c_str()) && !MainMenuController::checkIfDemoFileExists_4D1430(lvFilenameNoPrefix) && !MainMenuController::checkIfDemoFileExists_4D1430(lvFilename))
```
Comment (paulsapps) [inline Source/AliveLib/MainMenu.cpp:2070]: can't checkIfDemoFileExists_4D1430 be changed to use const char* ?

File: Source/AliveLib/MainMenu.cpp [hunk 2]
```diff
@@ -2062,8 +2064,15 @@ signed int MainMenuController::LoadDemo_Update_4D1040(DWORD)
         strcpy(&lvFilename[7], sPathData_559660.paths[levelId].field_22_lvl_name_cd);
         auto lvFilenameNoPrefix = &lvFilename[7];
 
-        while (!MainMenuController::checkIfDemoFileExists_4D1430(lvFilenameNoPrefix) && !MainMenuController::checkIfDemoFileExists_4D1430(lvFilename))
+#if BEHAVIOUR_CHANGE_SUB_DATA_FOLDERS
+		std::string subFolderPath(&lvFilenameNoPrefix[1]);
```
Comment (paulsapps) [inline Source/AliveLib/MainMenu.cpp:2068]: All this stl usage will make PS2 ports and others a lot harder to do :(

File: Source/AliveLib/Game.cpp [hunk 3]
```diff
@@ -361,7 +361,7 @@ EXPORT void CC Main_ParseCommandLineArguments_494EA0(const char* /*pCmdLineNotUs
         }
 
 #if DEVELOPER_MODE
-        if (strstr(pCommandLine, "-debug"))
+        //if (strstr(pCommandLine, "-debug"))
```
Comment (paulsapps) [inline Source/AliveLib/Game.cpp:364]: any reason this is removed ?? Delete if not used?

File: Source/AliveLib/Input.cpp [hunk 4]
```diff
@@ -1477,22 +1473,43 @@ EXPORT void Input_InitJoyStick_460080()
...
                     sJoystickEnabled_5C2EF4 = true;
+					sJoystickEnabled_5C9F70 = 1;
                     pSDLControllerHaptic = SDL_HapticOpenFromJoystick(SDL_GameControllerGetJoystick(pSDLController));
                     if (SDL_HapticRumbleInit(pSDLControllerHaptic) < 0)
                     {
                         printf("Warning: Unable to initialize rumble! SDL Error: %s\n", SDL_GetError());
                     }
                     strncpy(sGamePadStr_55E85C, SDL_GameControllerName(pSDLController), 32u);
+					
+					DEV_CONSOLE_PRINTF("Active Controller: %s", SDL_GameControllerName(pSDLController));
```
Comment (paulsapps) [inline Source/AliveLib/Input.cpp:1510]: tabs

File: Source/AliveLib/Input.cpp [hunk 5]
```diff
@@ -1477,22 +1473,43 @@ EXPORT void Input_InitJoyStick_460080()
...
+	
+    if (!bInitJoy)
     {
         DEV_CONSOLE_PRINTF("SDL GamePads: %i", SDL_NumJoysticks());
         for (int i = 0; i < SDL_NumJoysticks(); ++i) {
             if (SDL_IsGameController(i)) {
                 pSDLController = SDL_GameControllerOpen(i);
                 if (pSDLController) {
                     sJoystickEnabled_5C2EF4 = true;
+					sJoystickEnabled_5C9F70 = 1;
```
Comment (paulsapps) [inline Source/AliveLib/Input.cpp:1502]: tabs

File: Source/AliveLib/Io.cpp [hunk 6]
```diff
@@ -26,10 +29,29 @@ IO_FileHandleType IO_Open(const char* fileName, const char * mode)
         fileName += 2;
     }
 
+	std::string fixedPath(fileName);
+
+	// lowercase our string.
+	std::transform(fixedPath.begin(), fixedPath.end(), fixedPath.begin(),
```
Comment (paulsapps) [inline Source/AliveLib/Io.cpp:35]: if this is unconditional its probably gonna break mac/linux

File: Source/AliveLib/PauseMenu.cpp [hunk 7]
```diff
@@ -1011,6 +1016,85 @@ void PauseMenu::RestartPath()
...
+					// Cant load save. Play an error sound.
+					SFX_Play_46FBA0(17u, 30, 2400);
+				}
+			}
+
+			} });
+	}
+}
+
+CustomPauseMenu consoleSaveMenu(&consoleSaveMenuItems, "Save Game");
```
Comment (paulsapps) [inline Source/AliveLib/PauseMenu.cpp:1095]: var can be static cause its file scope global

File: Source/AliveLib/PauseMenu.cpp [hunk 8]
```diff
@@ -1011,6 +1016,85 @@ void PauseMenu::RestartPath()
 const char kArrowChar = 3;
 ALIVE_ARY(1, 0x55E398, char, 2, sArrowStr_55E398, { kArrowChar, 0 });
 
+#if CONSOLE_SAVE_MENU
+const int saveSlotCount = 8;
+
+std::string saveSlotTexts[saveSlotCount];
```
Comment (paulsapps) [inline Source/AliveLib/PauseMenu.cpp:1022]: these globals can be static

File: Source/AliveLib/PauseMenu.cpp [hunk 9]
```diff
@@ -1011,6 +1016,85 @@ void PauseMenu::RestartPath()
...
+std::vector<CustomPauseMenuItem> consoleSaveMenuItems;
+bool savePageToSave = true;
+void save_game_slot(int slot)
```
Comment (paulsapps) [inline Source/AliveLib/PauseMenu.cpp:1025]: probably the whole OG menu should be redone so that we don't need per platform menus

Comment (mlgthatsme) [reply Source/AliveLib/PauseMenu.cpp:1025]: agreed.

File: Source/AliveLib/MainMenu.cpp (hunk 2 above)
Comment (mlgthatsme) [reply Source/AliveLib/MainMenu.cpp:2068]: fair point. ill use some older techniques instead

File: Source/AliveLib/Io.cpp (hunk 6 above)
Comment (mlgthatsme) [reply Source/AliveLib/Io.cpp:35]: batch file to lowercase all the file names? thats what ive done. unless theres a way to open files case insenstive.

Comment (paulsapps) [reply Source/AliveLib/Io.cpp]: There is - but even still its gonna cause a regression if its merged as-is. Old alive proj has code to do a dir listing and then pick a file that matches any case.

File: Source/AliveLib/Movie.cpp [hunk 10]
```diff
@@ -408,7 +408,10 @@ EXPORT char CC DDV_Play_Impl_4932E0(const char* pMovieName)
 
             if (sFrameInterleaveNum_5CA23C > 15)
             {
-                if (Input_IsVKPressed_4EDD40(VK_ESCAPE) || Input_IsVKPressed_4EDD40(VK_RETURN))
+                sInputObject_5BD4E0.Update_45F040();
+                if (Input_IsVKPressed_4EDD40(VK_ESCAPE) || Input_IsVKPressed_4EDD40(VK_RETURN)
+                    || sInputObject_5BD4E0.isPressed(InputCommands::eFartOrRoll) || sInputObject_5BD4E0.isPressed(InputCommands::ePause)
```
Comment (d3nwah) [inline Source/AliveLib/Movie.cpp:413]: This is just wrong. farting or rolling actions shouldn't be connected with the movie playback

File: Source/AliveLib/logger.hpp [hunk 11]
```diff
@@ -15,13 +16,22 @@
 
 #define LOGGING 1
 
-#ifdef LOGGING
+#if LOGGING
+#if DISABLE_EASYLOGGING
+#define TRACE_ENTRYEXIT 
+#define LOG_TRACE(msg) //printf("[T] %s\n", msg)
```
Comment (paulsapps) [inline Source/AliveLib/logger.hpp:22]: remove comments?

File: Source/AliveLib/Movie.cpp (hunk 10 above)
Comment (paulsapps) [reply Source/AliveLib/Movie.cpp:413]: Hmm - what does PSX do? Given there isn't escape/enter keys

**Comment by paulsapps** [comment]
This isn't gonna be fixed or merged anytime soon so closing for now to avoid clogging up the PR list.

**Pull request closed without merging**